system.go

```diff
@@ -388,22 +387,39 @@ func (c *Client) DeleteSystemRecursive(name string, recursive bool) error {
 	return err
 }
 
-// CreateInterface creates network interfaces in Cobbler
-func (s *System) CreateInterface(name string, iface Interface) error {
+func makeInterfaceOptionsMap(name string, iface Interface) map[string]interface{} {
 	i := structs.Map(iface)
 	nic := make(map[string]interface{})
 	for key, value := range i {
 		attrName := fmt.Sprintf("%s-%s", key, name)
 		log.Printf("[DEBUG] Cobblerclient: setting interface attr %s to %s", attrName, value)
 		nic[attrName] = value
 	}
+	return nic
+}
+
+func (c *Client) ModifyInterface(systemID string, nic map[string]interface{}) error {
+	editUncasted, err := c.Call("modify_system", systemID, "modify_interface", nic, c.Token)
+	ok, editSuccessful := editUncasted.(bool)
+	if !ok {
+		return fmt.Errorf("editing interfaces of system %s failed due to an invalid return value of the server", systemID)
+	}
+	if !editSuccessful {
+		return fmt.Errorf("editing interface of system %s failed", systemID)
+	}
+	return err
+}
```

Comment:
I prefer the way you did it in the `RenameNetworkInterface` function, not immediately handling the error of the API call and checking the good values can be dangerous. Also then you can return nil at the end which helps with clarity